storage/src/mmr/mutable.rs

```diff
@@ -0,0 +1,512 @@
+//! An in-memory authenticatable & updatable key-value store.
+//!
+//! A _mutable MMR_ is an authenticatable and updatable key-value store based on an [Mmr] over all
+//! updates called the _update tree_, and an authenticatable [Bitmap] that indicates which leaves in
+//! the update tree correspond to the latest value of each key. For example, if the same key K was
+//! updated at positions X, Y, and Z in that order, then the bitmap will store 0s for the bits
+//! corresponding to X and Y, and a 1 for the bit corresponding to Z. The root hash of the mutable
```

Comment:
I suppose its premature to suggest we mention keeping the Bitmap around is optional (until that is actually supported)?

Reply:
Yeah wanted to add the optionality in a later pass.

Reply:
From our IRL conversaion: Putting this PR back in draft to separate out the bitmap part.

Will also make the "update type" (eg. delete vs update) as part of the digest.